R/exactMatchingCohort.R

```diff
@@ -88,14 +91,45 @@ exactMatchingCohort <- function(cohort, matchSex = TRUE, matchYearBirth = TRUE,
     ) %>%
     dplyr::left_join(
       cohort %>%
-        dplyr::select("cases_id" = "subject_id", "index_date" = "cohort_start_date"),
+        dplyr::select("cases_id" = "subject_id", "cohort_start_date","cohort_end_date"),
       by = "cases_id"
     ) %>%
     dplyr::filter(
-      .data$index_date >= .data$observation_period_start_date,
-      .data$index_date <= .data$observation_period_end_date
+      .data$cohort_start_date >= .data$observation_period_start_date,
+      .data$cohort_start_date <= .data$observation_period_end_date
+    ) %>%
+    dplyr::ungroup() %>%
+    dplyr::select("cases_id", "controls_id", "cohort_definition_id", "cohort_start_date","cohort_end_date")
+
+
+  # Format output
+  cohort_ref <- matches %>%
+    dplyr::select(-.data$controls_id) %>%
+    dplyr::rename("subject_id" = "cases_id") %>%
+    dplyr::union_all(
+      matches %>%
+        dplyr::select(-.data$cases_id) %>%
+        dplyr::rename("subject_id" = "controls_id") %>%
+        dplyr::mutate(cohort_definition_id = .data$cohort_definition_id + 0.5)
     ) %>%
-    dplyr::select("cases_id", "controls_id", dplyr::any_of(matchCols))
-    return(matches)
+    CDMConnector::compute_query()
+
+  cohort_set_ref <- cohort_set_ref %>%
```

Comment:
cohortSet:
- cohort_definition_id
- cohort_name
- target_cohort_name # tots igual
- target_cohort_id
- mach_status: target/matched
- match_sex # tots igual TRUE/FALSE
- match_year_of_birth # tots igual TRUE/FALSE

Reply:
what do you mean with "target_cohort_id"?